engine/classes/Elgg/Http/OkResponse.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Elgg\Http;
+
+use InvalidArgumentException;
+
+/**
+ * Response builder
+ * 
+ * @access private
+ */
+class OkResponse implements ResponseBuilder {
+
+	/**
+	 * @var string
+	 */
+	protected $content;
+
+	/**
+	 * @var int
+	 */
+	protected $status_code;
+
+	/**
+	 * @var string
+	 */
+	protected $forward_url;
+
+	/**
+	 * @var array
+	 */
+	protected $headers;
+
+	/**
+	 * Constructor
+	 *
+	 * @param mixed  $content     Response data
+	 * @param int    $status_code HTTP status code
+	 * @param string $forward_url Forward URL
+	 */
+	public function __construct($content = '', $status_code = ELGG_HTTP_OK, $forward_url = null) {
+		$this->setContent($content);
+		$this->setStatusCode($status_code);
+		$this->setForwardURL($forward_url);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function setContent($content = '') {
+		if (!is_scalar($content) && !is_array($content)) {
+			throw new InvalidArgumentException(__METHOD__ . ' expects $content as a scalar value or array');
+		}
+		$this->content = $content;
+		return $this;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getContent() {
+		return $this->content;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function setStatusCode($status_code = 200) {
+		$this->status_code = (int) $status_code;
+		return $this;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getStatusCode() {
+		return $this->status_code;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function setForwardURL($forward_url = REFERRER) {
+		$this->forward_url = $forward_url;
+		return $this;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getForwardURL() {
+		return $this->forward_url;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function setHeaders(array $headers = []) {
+		$this->headers = $headers;
+		return $this;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getHeaders() {
+		return (array) $this->headers;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function isInformational() {
+		return $this->status_code >= 100 && $this->status_code <= 199;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function isOk() {
+		return $this->status_code >= 200 && $this->status_code <= 299;
```

Comment:
We need to consider what Symfony is doing for redirects at least, eitherwise we end up with fatal errors.